### PR TITLE
Add teletubbies list

### PR DESCRIPTION
**Pull request by whitecollin**
-add teletubbies component to teletubbies page
-update teletubbies page

### CODE REVIEW

File: src/api/index.js [hunk 1]
```diff
@@ -18,3 +18,19 @@ export const getCurrentWeatherData = async () => {
...
+    const response=await fetch('teletubbies.json',{
+       headers: {
+         'Content-Type': 'application/json',
+         'Accept': 'application/json'
+       }
+     })
+     const myJson = await response.json();
+     return myJson;
+  }
+  catch (err){
```
Comment (codeLover1126) [inline src/api/index.js:33]: This `catch` block should be considered on the part where `getData` is going to be used. I mean, `src/pages/Teletubbies/index.js`.

File: src/pages/Teletubbies/index.js [hunk 2]
```diff
@@ -1,9 +1,16 @@
...
 import { Container, Typography, Box } from '@mui/material'
 import Teletubbie from 'components/Teletubbie'
+import { getData } from 'api'
 
 const Teletubbies = () => {
-  const [data, setData] = useState([])
+  const [data, setData] = useState([]);
+  useEffect(() => {
+    (async function () {
+      const data = await getData()
```
Comment (codeLover1126) [inline src/pages/Teletubbies/index.js:12]: what if `getData` is not working? please use `catch` here.

File: src/pages/Teletubbies/index.js [hunk 3]
```diff
@@ -5,18 +5,34 @@ import { getData } from 'api'
...
+      if (
+        window.innerHeight + window.scrollY >=
+        document.body.offsetHeight - 1000
+      ) {
+        setOffset(offset+20);
+      }
+    }
+    window.addEventListener('scroll', handleScroll)
+    return () => {
+      window.removeEventListener('scroll', handleScroll)
```
Comment (codeLover1126) [inline src/pages/Teletubbies/index.js:27]: Nice!

**Comment by codeLover1126** [comment]
Also, keep the PR title more professional.

File: src/api/index.js [hunk 4]
```diff
@@ -18,3 +18,19 @@ export const getCurrentWeatherData = async () => {
...
+    console.log(err);
+  }
+ }
```
[the remark this answers is not in the document]
Comment (whitecollin) [reply src/api/index.js]: fixed it

File: src/api/index.js [hunk 5]
```diff
@@ -18,3 +18,19 @@ export const getCurrentWeatherData = async () => {
     throw error;
   }
 };
+
+export const getData = async() => {
+  try{
+    const response=await fetch('teletubbies.json',{
+       headers: {
```
[the remark this answers is not in the document]
Comment (whitecollin) [reply src/api/index.js]: fixed it

File: src/api/index.js (hunk 1 above)
Comment (whitecollin) [reply src/api/index.js]: fixed it

File: src/pages/Teletubbies/index.js (hunk 2 above)
Comment (whitecollin) [reply src/pages/Teletubbies/index.js]: fixed ti

File: src/pages/Teletubbies/index.js [hunk 6]
```diff
@@ -5,13 +5,18 @@ import { getData } from 'api'
 
 const Teletubbies = () => {
   const [data, setData] = useState([]);
```
Comment (codeLover1126) [inline src/pages/Teletubbies/index.js:7]: it's better to name `teletubbies` instead of just `data`.

File: src/api/index.js [hunk 7]
```diff
@@ -20,7 +20,6 @@ export const getCurrentWeatherData = async () => {
 };
 
 export const getData = async() => {
```
Comment (codeLover1126) [inline src/api/index.js:22]: what about naming `getTeletubbies`?

File: src/api/index.js [hunk 8]
```diff
@@ -29,8 +28,4 @@ export const getData = async() => {
     })
     const myJson = await response.json();
```
Comment (codeLover1126) [inline src/api/index.js:29]: not good naming: `myJson`.

File: src/api/index.js [hunk 9]
```diff
@@ -20,7 +20,6 @@ export const getCurrentWeatherData = async () => {
 };
 
 export const getData = async() => {
-  try{
     const response=await fetch('teletubbies.json',{
```
Comment (codeLover1126) [inline src/api/index.js:23]: please add spaces before and after '='.

File: src/pages/Teletubbies/index.js [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+    return () => {
+      window.removeEventListener('scroll', handleScroll)
+    }
+  }, [offset])
+  return (
+    <Container maxWidth='xl'>
+      <Typography variant='h1' sx={{ marginTop: '30px' }}>
+        Teletubbies
+      </Typography>
+      {data.slice(0,20+offset)
```
Comment (codeLover1126) [inline src/pages/Teletubbies/index.js:40]: Missing spaces before and after ','.

**Review by codeLover1126 — CHANGES_REQUESTED** [review]

**Comment by codeLover1126** [comment]
I think it's good to name PR `Add teletubbies list`. Please change the description as a list.

File: src/pages/Teletubbies/index.js (hunk 6 above)
Comment (whitecollin) [reply src/pages/Teletubbies/index.js:7]: fixed it

File: src/api/index.js (hunk 7 above)
Comment (whitecollin) [reply src/api/index.js:22]: fixed it

**Pull request merged**